lib/protocol/collection/notes_collection.ts

```diff
@@ -0,0 +1,94 @@
+import { SNNote, SNTag, SNSmartTag, SNPredicate, SNItem, ContentType } from '../../models'
+import { ItemCollection, CollectionSort, SortDirection } from "./item_collection";
+import { UuidString } from "@Lib/types";
+
+export class NotesCollection {
```

Comment:
Why did you decide against subclassing ItemCollection?

Reply:
ItemCollection handles multiple content types. To subclass it would have meant that NotesCollection would be expected to handle arbitrary content types, which doesn't make sense and would likely lead to misunderstandings down the road. 
By using composition (NotesCollection has an ItemCollection) NotesCollection becomes a specialized view of ItemCollection, which makes it easier to pull out or change, and easier to understand (no implicit parent behavior).

Reply:
Maybe this needs another name actually, like ItemCollectionNotesView.

Reply:
Other alternatives I considered were:
1) to have ItemCollection include NotesCollection and leave all the sorting and filtering to it. I was worried of bloating the itemCollection class with special cases however.
2) to have NotesCollection be a MutableCollection subclass and handle sorting itself. This would involve extracting the sorting functionality out of ItemCollection and into generic functions that NotesCollection could call into, and not feeding ItemCollection with notes anymore, to prevent unnecessary computations.

I was strongly tempted by 2) and the reason I went with my current implementation is that it seemed less disruptive and kept one single source of truth vs two (NotesCollection for notes vs ItemCollection for everything else).